src/slowmokit/methods/metrics/silhouette_score.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file methods/metrics/silhouette_score.cpp
+ *
+ * Implementation of the Silhouette Score main program
+ */
+
+#include "silhouette_score.hpp"
+template<class T>
+double SilhouetteScore(std::vector<std::vector<T>> x, std::vector<int> y,
+                       int numClusters, std::string typeDist)
+{
+  std::vector<std::vector<double>> distances(x.size(),
+                                             std::vector<double>(x.size()));
+  for (int i = 0; i < x.size(); i++)
+  { // iterating over each x[i]
+    for (int j = 0; j < x.size(); j++)
+    { // iterating over each x[i]
+      if (i == j)
+      {
+        distances[i][j] = 0.0; // initialize distance=0 if computing distance
+                               // between same values
+      }
+      else
+      {
+        if (typeDist == "euclidean")
+        { // euclidean distance
+          for (int k = 0; k < x[0].size(); k++)
+          {
+            distances[i][j] += pow((x[i][k] - x[j][k]),
+                                   2); // x[i][k]-> i=point,k=1-d value of point
+          }
+          distances[i][j] = pow(distances[i][j], 0.5);
+        }
+        else if (typeDist == "manhattan")
+        {
+          for (int k = 0; k < x[0].size(); k++)
+          {
+            distances[i][j] += (x[i][k] - x[j][k]);
+          }
+        }
+      }
+    }
+  }
+
+  std::vector<double> intraClusters(
+      x.size(), 0.0); // Computing intraclusters distances of each point
+  std::vector<double> crossClusters(
+      x.size()); // Minimum Distance of each point to other clusters
+  for (int i = 0; i < x.size(); i++)
+  {
+    int sumnum = 0;
```

Comment:
```suggestion
    int sumNum = 0;
```
change to camel case